### PR TITLE
NPUW: Disable AVX2 code with ENABLE_AVX2=OFF 

### PR DESCRIPTION
**Pull request by eshiryae**
### Details:
 - Disable AVX2 code with ENABLE_AVX2=OFF

### Tickets:
 - E-141645

### CODE REVIEW

**Comment by dmatveev** [comment]
Thanks @eshiryae for the quick turnaround! It seems you need to run `clang_format_fix_all` in your branch
@esmirno can you please review the changes? I'll come up with review later

**Comment by esmirno** [comment]
LGTM certain avx instruction requires F16C module,  while it will be hard to differentiate, and more over openvino doesn't have this level of granularity in settings, so as first step i think this is ok.

**Review by esmirno — APPROVED** [review]

File: src/plugins/intel_npu/src/plugin/npuw/util.cpp
Comment (smirnov-alexey) [inline src/plugins/intel_npu/src/plugin/npuw/util.cpp:90]: Something went wrong with indentation here

Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/util.cpp:90]: Or is it just a clang-format thing?

**Review by smirnov-alexey — APPROVED** [review]

File: src/plugins/intel_npu/src/plugin/npuw/util.cpp
Comment (eshiryae) [reply src/plugins/intel_npu/src/plugin/npuw/util.cpp:91]: It's clang-format thing

**Comment by esmirno** [comment]
please rebase this PR, once Keras fix merged :  https://github.com/openvinotoolkit/openvino/pull/26912

File: src/plugins/intel_npu/src/plugin/npuw/util.cpp
Comment (ilya-lavrenov) [inline src/plugins/intel_npu/src/plugin/npuw/util.cpp:504]: why not `NPUW_ASSERT` ?

File: src/plugins/intel_npu/src/plugin/npuw/util.cpp
Comment (ilya-lavrenov) [inline src/plugins/intel_npu/src/plugin/npuw/util.cpp:67]: who defines `HAVE_AVX2` for this file if AVX2 is enabled in cmake?
In CPU source files, we use `cross_compiled_file` which defines `HAVE_<instruction set>` automatically based on compiled architectures. Looks like in case of NPU this code will always be disabled, right?

Comment (dmatveev) [reply src/plugins/intel_npu/src/plugin/npuw/util.cpp:67]: it seems if that code worked for @eshiryae, somehow it is defined when `util.cpp` is compiled - otherwise an exception would be thrown when running a model with DCOFF configured. Anyways let's wait for Ekaterina's reply

File: src/plugins/intel_npu/src/plugin/npuw/util.cpp
Comment (eshiryae) [reply src/plugins/intel_npu/src/plugin/npuw/util.cpp]: Thanks for suggestion, fixed, but decided to use OPENVINO_THROW (as don't need to check any expression here)

File: src/plugins/intel_npu/src/plugin/npuw/util.cpp
Comment (eshiryae) [reply src/plugins/intel_npu/src/plugin/npuw/util.cpp]: Added `cross_compiled_file` using for AVX2 specific code. 
Yes, basically in previous implementation this code was always disabled and exception was thrown when model is running. Now  HAVE_AVX2 should be enabled correctly, but see other behavior - failed with exception, even though file were compiled with avx2 flags. @ilya-lavrenov could you please help to understand what is missing?

Comment (eshiryae) [reply src/plugins/intel_npu/src/plugin/npuw/util.cpp]: Found that my Linux VM doesn't have AVX2 support, checked on other client - everything is working.

**Review by dmatveev — CHANGES_REQUESTED** [review]
Let's discuss if needed.

If you want to dispatch over top-level `unpack`, please keep the old `unpack` back somewhere in util and call `XARCH::unpack` from it.

**Review by dmatveev — APPROVED** [review]
Great!

**Review by smirnov-alexey — APPROVED** [review]

**Review by esmirno — APPROVED** [review]

**Pull request merged**
